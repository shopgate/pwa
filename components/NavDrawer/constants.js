export const QUICKLINKS_MENU = 'quicklinks';
