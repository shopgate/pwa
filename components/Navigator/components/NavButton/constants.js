export const NAV_STATE_INDEX = 'NAV_STATE_INDEX';
export const NAV_STATE_BACK = 'NAV_STATE_BACK';
