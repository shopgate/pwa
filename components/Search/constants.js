export const TOGGLE_SEARCH = 'TOGGLE_SEARCH';
