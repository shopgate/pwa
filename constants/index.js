/**
 * Defines the maximum width the page content will extend.
 */
export const MAX_DESKTOP_WIDTH = 1300;

/**
 * Defines the width of the menu bar in desktop mode.
 */
export const DESKTOP_MENU_BAR_WIDTH = 208;

export const DESKTOP_MENU_BAR_HEIGHT = 64;
