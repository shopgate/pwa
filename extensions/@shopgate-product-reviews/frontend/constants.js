/**
 * Value to recalculate the rating average.
 * @type {number}
 */
export const RATING_SCALE_DIVISOR = 20;

/**
 * The number of stars to show.
 * @type {number}
 */
export const NUMBER_OF_STARS = 5;
