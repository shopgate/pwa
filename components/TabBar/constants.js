import { BEFORE, AFTER } from '@shopgate/pwa-common/constants/Portals';

export const TAB_HOME = 'home';
export const TAB_BROWSE = 'browse';
export const TAB_CART = 'cart';
export const TAB_MORE = 'more';
export const TAB_NONE = 'none';
export const TAB_FAVORITES = 'favorites';

const HOME_ICON = `${TAB_HOME}-icon`;
const BROWSE_ICON = `${TAB_BROWSE}-icon`;
const CART_ICON = `${TAB_CART}-icon`;
const MORE_ICON = `${TAB_MORE}-icon`;
const FAVORITES_ICON = `${TAB_FAVORITES}-icon`;

export const TAB_BAR = 'tab-bar';
export const TAB_BAR_BEFORE = `${TAB_BAR}.${BEFORE}`;
export const TAB_BAR_AFTER = `${TAB_BAR}.${AFTER}`;

export const TAB_BAR_HOME_BEFORE = `${TAB_BAR}.${TAB_HOME}.${BEFORE}`;
export const TAB_BAR_HOME = `${TAB_BAR}.${TAB_HOME}`;
export const TAB_BAR_HOME_AFTER = `${TAB_BAR}.${TAB_HOME}.${AFTER}`;

export const TAB_BAR_BROWSE_BEFORE = `${TAB_BAR}.${TAB_BROWSE}.${BEFORE}`;
export const TAB_BAR_BROWSE = `${TAB_BAR}.${TAB_BROWSE}`;
export const TAB_BAR_BROWSE_AFTER = `${TAB_BAR}.${TAB_BROWSE}.${AFTER}`;

export const TAB_BAR_CART_BEFORE = `${TAB_BAR}.${TAB_CART}.${BEFORE}`;
export const TAB_BAR_CART = `${TAB_BAR}.${TAB_CART}`;
export const TAB_BAR_CART_AFTER = `${TAB_BAR}.${TAB_CART}.${AFTER}`;

export const TAB_BAR_MORE_BEFORE = `${TAB_BAR}.${TAB_MORE}.${BEFORE}`;
export const TAB_BAR_MORE = `${TAB_BAR}.${TAB_MORE}`;
export const TAB_BAR_MORE_AFTER = `${TAB_BAR}.${TAB_MORE}.${AFTER}`;

export const TAB_BAR_FAVORITES_BEFORE = `${TAB_BAR}.${TAB_FAVORITES}.${BEFORE}`;
export const TAB_BAR_FAVORITES = `${TAB_BAR}.${TAB_FAVORITES}`;
export const TAB_BAR_FAVORITES_AFTER = `${TAB_BAR}.${TAB_FAVORITES}.${AFTER}`;

export const TAB_BAR_HOME_ICON = `${TAB_BAR}.${HOME_ICON}`;
export const TAB_BAR_BROWSE_ICON = `${TAB_BAR}.${BROWSE_ICON}`;
export const TAB_BAR_CART_ICON = `${TAB_BAR}.${CART_ICON}`;
export const TAB_BAR_MORE_ICON = `${TAB_BAR}.${MORE_ICON}`;
export const TAB_BAR_FAVORITES_ICON = `${TAB_BAR}.${FAVORITES_ICON}`;

export const ENABLE_TAB_BAR = 'ENABLE_TAB_BAR';
export const DISABLE_TAB_BAR = 'DISABLE_TAB_BAR';

export const SHOW_TAB_BAR = 'SHOW_TAB_BAR';
export const HIDE_TAB_BAR = 'HIDE_TAB_BAR';
