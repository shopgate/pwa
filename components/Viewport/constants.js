import { BROWSE_PATH } from 'Pages/Browse/constants';

export const a11yNavEntries = [
  {
    title: 'navigation.categories',
    link: BROWSE_PATH,
  },
];
