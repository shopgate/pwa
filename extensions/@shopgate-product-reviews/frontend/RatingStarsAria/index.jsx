import React from 'react';
import PropTypes from 'prop-types';
import appConfig from '@shopgate/pwa-common/helpers/config';
import { RATING_SCALE_DIVISOR, NUMBER_OF_STARS } from '../constants';
import connect from './connector';

const { hasReviews } = appConfig;

/**
 * The RatingStarsAria component.
 * @param {Object} props The component props.
 * @param {Object} context The component context.
 * @return {JSX}
 */
const RatingStarsAria = ({ product }, context) => {
  if (!hasReviews || !product || !product.rating || product.rating.average === 0) {
    return null;
  }

  const { __ } = context.i18n();
  const label = __('reviews.rating_stars', {
    rate: product.rating.average / RATING_SCALE_DIVISOR,
    maxRate: NUMBER_OF_STARS,
  });

  return <span aria-label={label} />;
};

RatingStarsAria.propTypes = {
  product: PropTypes.shape().isRequired,
};

RatingStarsAria.contextTypes = {
  i18n: PropTypes.func,
};

export default connect(RatingStarsAria);
