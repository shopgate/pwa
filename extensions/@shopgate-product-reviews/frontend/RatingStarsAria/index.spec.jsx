import React from 'react';
import { shallow } from 'enzyme';
import { NUMBER_OF_STARS } from '../constants';
import RatingStarsAria from './index';

jest.mock('./connector', () => obj => obj);

const mockTranslate = jest.fn(key => key);

const mockRenderOptions = {
  context: {
    i18n: () => ({
      __: mockTranslate,
    }),
  },
};

const product = {
  rating: {
    average: 70,
  },
};

describe('<RatingStars />', () => {
  it('should render as expected for 3.4 stars', () => {
    const wrapper = shallow(<RatingStarsAria product={product} />, mockRenderOptions);
    expect(wrapper).toMatchSnapshot();
    expect(mockTranslate).toHaveBeenCalledWith('reviews.rating_stars', {
      rate: 3.5,
      maxRate: NUMBER_OF_STARS,
    });
  });
});
