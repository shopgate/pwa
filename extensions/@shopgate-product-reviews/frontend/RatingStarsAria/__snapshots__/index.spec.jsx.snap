// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingStars /> should render as expected for 3.4 stars 1`] = `
<span
  aria-label="reviews.rating_stars"
/>
`;
